src/covid_shared/cli_tools/decorators.py

```diff
@@ -48,3 +50,109 @@ def _wrapped(*args, **kwargs):
         return _wrapped
 
     return _pass_run_metadata
+
+
+def add_r_singularity_option():
+    """
+    Decorator to set R environment to use in a CLI command.
+    """
+    def wrapper(entry_point: types.FunctionType):
+        entry_point = click.option(
+            '--r-singularity-image', 'r_singularity_image',
+            type=click.Path(file_okay=True),
+            default=R_SINGULARITY_IMAGE_PATH,
+            help=(f"Path to a specific R singularity. Can pass 'None' to use active conda environment. "
+                  f"By default uses {R_SINGULARITY_IMAGE_PATH}.")
+        )(entry_point)
+
+        return entry_point
+    return wrapper
+
+
+def add_create_directory_options(default_output_root: Path):
+    """
+    Decorator to set CLI command options required when producing outputs.
+    """
+    def wrapper(entry_point: types.FunctionType):
+
+        entry_point = click.option(
+            "-o", "--output-root",
+            type=click.Path(file_okay=False),
+            default=default_output_root,
+            help=(f"Directory to outputs. Defaults to {default_output_root}/YYYY_MM_DD.VV for today's date and the "
+                  f"newest uncreated version")
```

Comment:
my linter complains about f-strings that don't do any substitution.

Reply:
I'm super sloppy about this.  But it is right.